x-pack/plugins/maps/public/classes/util/can_skip_fetch.ts

```diff
@@ -55,10 +55,12 @@ export async function canSkipSourceUpdate({
   source,
   prevDataRequest,
   nextMeta,
+  considerSpatialParameters,
```

Comment:
Is considerSpatialParameters even needed? I think you should update the URL when the extent changes. Let me provide an example to explain
The map has 2 layers:
  MVT document layer
  document layer

The time picker is set to "last 15 minutes".

Both layers load and are in sync.

5 minutes passes on the clock. The user pans the map so `updateDueToExtentChange` is true

 MVT document layer now using stall time ranges compared to the other layer and is really showing "20 minutes to 5 minutes" ago. Its now out of sync with document layer

Reply:
we should discuss. 

there's two parts to this imho:

- wrt `considerSpatialParameters`: the {x}/{y}/{z} structure fully captures the spatial params. mapbox-gl will appropriately load the correct based on the pan-zoom movements of the user. I am not sure Maps needs to introduce another layer of "spatial awareness" on top of this. hence why I think we should ignore it in the canSkip-check for mvbt.
- refreshing based on time: one of the key advantages of vector-tiles is the ability to cache tiles client-side, for reuse within a single app-session. if Maps refreshes the mvt-source, each time an extent changes, the end users essentially loses that capability. This is exactly the sort of trade-off that a user needs to make between whether to use frequent refreshes (using json), or use mvt.

imho, there's indeed a larger challenge wrt mvt and _relative_ time ranges. mvt has the potential to make the mismatch more glaring. 

The issue for dealing with relative time is not only for mvt. It exists for all visualizations. _Kibana has the force-refresh button and the auto-refresh capability to deal with this_. To address stale-data issues due to relative time-selection, the user should use auto-update.

The mismatch that arises between multiple layer-configurations on the same source is real; I think this is more of a theoretical issue. There are multiple instances where mismatches between multiple layers can occur when there is a relative time-selection, based on how exactly the layer and source-settings diverge (e.g. apply global time turned on vs the other doesnt, layers are added at different times so the first layer will be stale but the second will be accurate, one has "update based on bounds" turned on/off vs the other doesnt, etc...).

I do agree there is potential to refresh tiles more aggressively, or have the artefacts of a source-change be less glaring. There may be more efficient ways of reloading tiles without replacing the entire source, by using these mapbox-gl capabilities of just updating source-params: https://github.com/mapbox/mapbox-gl-js/pull/8048

My proposal would be to address that in a separate PR (efficiently dealing with time-based auto-refresh)

Reply:
discussed offline: `considerSpatialParameters` is really icky looking, how it adds more nesting. Will move up the `extentAware` parameter to the function-signature and have the layer pass it in appropriately. Can remove getGeogridPrecision-awareness as well, since only heatmap-layers require it.